Assets/MRTK/SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs

```diff
@@ -407,41 +407,51 @@ public float SphereCastRadius
         public virtual void OnPreSceneQuery() { }
 
         /// <inheritdoc />
-        public virtual bool OnSceneQuery(LayerMask[] prioritizedLayerMasks, bool focusIndividualCompoundCollider, out MixedRealityRaycastHit hitInfo)
+        public virtual bool OnSceneQuery(LayerMask[] prioritizedLayerMasks, bool focusIndividualCompoundCollider, out MixedRealityRaycastHit hitInfo, out RayStep Ray, out int rayStepIndex)
         {
+            float rayStartDistance = 0;
             var raycastProvider = CoreServices.InputSystem.RaycastProvider;
-            switch (SceneQueryType)
+            for (int i = 0; i < Rays.Length; i++)
```

Comment:
I think the previous pattern is a bit more performant, since `SceneQueryType` shouldn't be changing during the iteration, so there's no need imo to run the `switch` on _every_ cycle of the loop.

Reply:
I can change it :) 
Overall it should be more performant (I can't believe I did redundant for loops the first time lol)

Reply:
done!